lib/handlers/quote/quote.ts

```diff
@@ -456,12 +457,18 @@ export class QuoteHandler extends APIGLambdaHandler<
       portionAmount: outputPortionAmount, // TODO: name it back to portionAmount
     } = swapRoute
 
-    const estimatedGasUsed = adhocCorrectGasUsed(preProcessedEstimatedGasUsed, chainId, isMobileRequest)
+    const estimatedGasUsed = adhocCorrectGasUsed(
+      preProcessedEstimatedGasUsed,
+      chainId,
+      isMobileRequest,
```

Comment:
i think this should happen inside the function (pass in the request source instead of parameters for each source)

Reply:
we do need `isMobileRequest` in that static function https://github.com/Uniswap/routing-api/blob/01dec1424e458cf0236b8167200f9b38ef9f92c1/lib/handlers/quote/quote.ts#L251, because we still have to exclude v2 routing from mobile quote request with version earlier than 1.24.

but I created a generic shouldCorrectGas boolean flag to pass, so that the `adhocCorrectGasUsed` function params shouldn't have to mutate again.

Reply:
@jsy1218 you can pass in the source and do the logic inside the `adhocCorrectGasUsed` and not outside of it because thats where the logic lives right? You can still run `isMobileRequest` but having the source alone should eb good enough for you to make a decision